Reflect the current roles when rolling Kafka nodes in KafkaRoller (#9686)
strimzi/strimzi-kafka-operator: documentation/modules/configuring/con-config-node-pool-roles.adoc

In this case, you create a new node pool with nodes that act only as brokers, and then reassign partitions from the dual-role nodes to the new brokers.
You can then switch the old node pool to a controller-only role.

> Or the other way around, when you want to move from node pools with controller-only and broker-only roles to a node pool that contains nodes that perform dual broker and controller roles.
> In this case, you can add the `broker` role to the existing controller-only node pool, reassign partitions from the broker-only nodes to the dual-role nodes, and then delete the broker-only node pool.

Reviewer: ```suggestion
You can also perform the reverse operation by moving from node pools with controller-only and broker-only roles to a node pool that contains nodes that perform dual broker and controller roles.
In this case, you add the `broker` role to the existing controller-only node pool, reassign partitions from the broker-only nodes to the dual-role nodes, and then delete the broker-only node pool.
```
